packages/explat-client-react-helpers/src/index.tsx

```diff
@@ -53,7 +53,10 @@ export default function createExPlatClientReactHelpers(
 			// eslint-disable-next-line react-hooks/exhaustive-deps
 		}, [] );
 
-		if ( experimentName !== previousExperimentName ) {
+		if (
+			experimentName !== previousExperimentName &&
+			! previousExperimentName.startsWith( 'explat_test' )
+		) {
 			const message = '[ExPlat] useExperiment: experimentName should never change between renders!';
```

Comment:
The `explat_test_aa_weekly_calypso` experiment is going to violate this constraint every week if you happen to be at the wrong Calypso screen at the wrong time 🙂 

The `useExperiment` hook could be rewritten to handle parameter name change gracefully. Let me know if you need any assistance with that.

Reply:
> The explat_test_aa_weekly_calypso experiment is going to violate this constraint every week if you happen to be at the wrong Calypso screen at the wrong time 🙂

The `! previousExperimentName.startsWith( 'explat_test' )` here should prevent that?

> The useExperiment hook could be rewritten to handle parameter name change gracefully. Let me know if you need any assistance with that.

The idea here is that once a component has rendered with an experiment, it should never change it's name, except for our internal automated A/A testing (tests we use to ensure the experiment system is not biased).

Reply:
I see, the `explat_test` prevents throwing the error :+1:

But still the `useEffect` hook that loads the assignment will not be run with the new (next week) name. It doesn't re-run on `experimentName` change. That's also desired in this case?

> The idea here is that once a component has rendered with an experiment, it should never change it's name

I know that in practice `useExperiment()` will always be called with the same `experimentName`. But it's a good practice in general for hooks to react correctly to parameter changes. It's similar to components reacting to a prop change, however unlikely. And the `useExperiment` hook could be easily adapted to make it work.

Reply:
> But still the useEffect hook that loads the assignment will not be run with the new (next week) name. It doesn't re-run on experimentName change. That's also desired in this case?

Ah! Nice catch!

> I know that in practice useExperiment() will always be called with the same experimentName. But it's a good practice in general for hooks to react correctly to parameter changes. It's similar to components reacting to a prop change, however unlikely. And the useExperiment hook could be easily adapted to make it work.

Yeah I guess it should do so.

Reply:
> That's also desired in this case?

Actually, on second thought, I think this I did have this in mind - we should wait until the next time this component is constructed rather than switch experiment mid-render. I think that is a nice consistent rule to have for all experiments but other than that the two ways to go about it are the same.